packages/app/ui/components/ChatMessage/ChatMessageActions/MessageActions.tsx

```diff
@@ -259,6 +282,115 @@ export async function handleAction({
       }
       triggerHaptic('success');
       return; // Early return to avoid double dismiss
+    case 'summarize': {
```

Comment:
This logic should be encapsulated within a `shared/store` model action file, not inlined here.